lib/rules/indent.js

```diff
@@ -429,13 +429,11 @@ module.exports = function(context) {
             nodeIndent = getNodeIndent(effectiveParent);
             if (parentVarNode && parentVarNode.loc.start.line !== node.loc.start.line) {
                 if (parent.type !== "VariableDeclarator" || parentVarNode === parentVarNode.parent.declarations[0]) {
-                    if (options.VariableDeclarator[parentVarNode.parent.kind] === 0 && (parent.type === "ObjectExpression" || parent.type === "ArrayExpression")) {
-                        nodeIndent = nodeIndent + indentSize;
-                    } else {
+                    if (parentVarNode.loc.start.line === effectiveParent.loc.start.line) {
                         nodeIndent = nodeIndent + (indentSize * options.VariableDeclarator[parentVarNode.parent.kind]);
+                    } else if (parent.type === "ObjectExpression" || parent.type === "ArrayExpression") {
+                        nodeIndent = nodeIndent + indentSize;
                     }
-                } else if (parent.loc.start.line !== node.loc.start.line && parentVarNode === parentVarNode.parent.declarations[0]) {
```

Comment:
This block was redundant since it would never be hit due to the repeated condition in the original `if` block.